fedimint-client/src/module/gen.rs

```diff
@@ -33,7 +34,36 @@ pub trait ClientModuleGen: ExtendsCommonModuleGen + Sized {
     type Config: TypedClientModuleConfig;
 
     /// Initialize a [`ClientModule`] instance from its config
-    async fn init(&self, cfg: Self::Config, db: Database) -> anyhow::Result<Self::Module>;
+    async fn init(
+        &self,
+        cfg: Self::Config,
+        db: Database,
+        instance_id: ModuleInstanceId,
+    ) -> anyhow::Result<Self::Module>;
+
+    /// Initialize a [`crate::module::PrimaryClientModule`] instance from its
+    /// config
+    ///
+    /// The default implementation returns an error, assuming that the module is
+    /// not a primary one. If it is the default impl has to be overridden as
```

Comment:
What does it mean to be a primary module?

Reply:
One used to generate change and "exact" inputs, to allow balancing out inputs and outputs when building txes.

Reply:
Lmk if the docs for `PrimaryClientModule` need to be improved, otherwise I think linking to it should be enough and we don't need to duplicate the explanation.